tests/Aql/AqlSharedExecutionBlockImplTest.cpp

```diff
@@ -0,0 +1,790 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2020-2020 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Michael Hackstein
+////////////////////////////////////////////////////////////////////////////////
+
+#include "gtest/gtest.h"
+
+#include "AqlItemBlockHelper.h"
+#include "ExecutorTestHelper.h"
+#include "Mocks/Servers.h"
+#include "WaitingExecutionBlockMock.h"
+
+#include "Aql/AllRowsFetcher.h"
+#include "Aql/AqlCallStack.h"
+#include "Aql/AqlItemBlock.h"
+#include "Aql/CountCollectExecutor.h"
+#include "Aql/ExecutionBlockImpl.h"
+#include "Aql/ExecutionEngine.h"
+#include "Aql/FilterExecutor.h"
+#include "Aql/IdExecutor.h"
+#include "Aql/ModificationExecutor.h"
+#include "Aql/ModificationExecutorInfos.h"
+#include "Aql/Query.h"
+#include "Aql/RegisterInfos.h"
+#include "Aql/SimpleModifier.h"
+#include "Aql/SingleRowFetcher.h"
+#include "Aql/SkipResult.h"
+#include "Aql/SortExecutor.h"
+#include "Aql/SortRegister.h"
+#include "Aql/UnsortedGatherExecutor.h"
+
+static_assert(GTEST_HAS_TYPED_TEST, "We need typed tests for the following:");
+
+using namespace arangodb;
+using namespace arangodb::aql;
+
+namespace arangodb {
+namespace tests {
+namespace aql {
+
+using InsertExecutor =
+    ModificationExecutor<SingleRowFetcher<BlockPassthrough::Disable>, InsertModifier>;
+
+/*
+ * TODO
```

Comment:
If still valid and open TODO, we should track them in our Kanban Board (if not already done).

Reply:
good catch.
let me check, this was my todo list while writing the tests, so either I still want to add those tests or I forgot to tick them off ;)

Reply:
ok checked, I forgot to tick them off, removed now.